src/components/create-or-edit-question/CreateOrEditQuestion.tsx

```diff
@@ -0,0 +1,192 @@
+import { ChangeEvent, MouseEvent, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import Divider from '@mui/material/Divider'
+import IconButton from '@mui/material/IconButton'
+import DeleteIcon from '@mui/icons-material/Delete'
+import CloseIcon from '@mui/icons-material/Close'
+import AddIcon from '@mui/icons-material/Add'
+import Checkbox from '@mui/material/Checkbox'
+import FormControlLabel from '@mui/material/FormControlLabel'
+import FormGroup from '@mui/material/FormGroup'
+import InputBase from '@mui/material/InputBase'
+import Radio from '@mui/material/Radio'
+import RadioGroup from '@mui/material/RadioGroup'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+import AppSelect from '~/components/app-select/AppSelect'
+
+import { styles } from '~/components/create-or-edit-question/CreateOrEditQuestion.styles'
+import { sortQuestions } from '~/components/create-or-edit-question/CreateOrEdit.constants'
+import { Answer, SizeEnum, TextFieldVariantEnum } from '~/types'
+
+const CreateOrEditQuestion = () => {
+  const [questionType, setQuestionType] = useState(sortQuestions[0])
+  const [question, setQuestion] = useState<string>('')
+  const [answers, setAnswers] = useState<Answer[]>([])
+  const [answer, setAnswer] = useState<string>('')
+
+  const { t } = useTranslation()
+
+  const isMultipleChoice = questionType.value === sortQuestions[0].value
+  const isOpenAnswer = questionType.value === sortQuestions[1].value
+  const isSingleChoice = questionType.value === sortQuestions[2].value
+
+  const setValue = (value: string) => {
+    const questionOption = sortQuestions.find((item) => item.value === value)
+
+    setQuestionType(questionOption ?? sortQuestions[0])
+  }
+
+  const handleQuestion = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuestion(event.target.value)
+  }
+
+  const sortOptions = sortQuestions.map(({ icon, title, value }) => ({
+    title: t(title),
+    value,
+    icon
+  }))
+
+  const handleOptionChange = (
+    event: ChangeEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    const updatedAnswers = [...answers]
+
+    if (isMultipleChoice) {
+      updatedAnswers[index].isCorrect = event.target.checked
+    } else if (isSingleChoice) {
+      updatedAnswers.forEach((answer, i) => {
+        answer.isCorrect = i === index
+      })
+    }
+
+    setAnswers(updatedAnswers)
+  }
+
+  const onChangeInput = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
```

Comment:
Should it have both HTMLInputElement and HTMLTextAreaElement?

Reply:
It can be two options, so I don't change